src/acceleration/BaseQNAcceleration.hpp

```diff
@@ -70,7 +71,8 @@ class BaseQNAcceleration : public Acceleration {
       int                     filter,
       double                  singularityLimit,
       std::vector<int>        dataIDs,
-      impl::PtrPreconditioner preconditioner);
+      impl::PtrPreconditioner preconditioner,
+      bool                    reduced = true);
```

Comment:
Currently the configuration is hard-coded to `reduced = true`. If we want to test the reduced variant we need to make it configurable first and we also need corresponding tests. I would actually prefer sticking to the reduced variant for now. According to [our paper](https://onlinelibrary.wiley.com/doi/epdf/10.1002/nme.6443) the non-reduced variant has little value (at least for the cases that we have studied). The way that the QN scheme is implemented we can activate the non-reduced variant at a later point in time if needed. (would also be a nice topic for a thesis to find a proper use case)

Reply:
I think it is configurable already? See the integration tests. I mainly meant for validating the implementation now.

Reply:
True. It is configurable. I would rather remove this option from this PR, get this PR merged, and add it the non-reduced variant later if needed. From my perspective rQN is sufficient and the non-reduced variant has no clear benefit. (sure: maybe there are examples that I do not have on the radar at the moment) If we also want to offer the non-reduced variant this requires additional validation and documentation for which I do not really have time at the moment. Factoring out the non-reduced QN into an independent PR should not be a big problem.

I'm not sure on what we agreed originally but having the non-reduced variant as a feature that we add later if needed sounds to me like the right approach to get rQN merged soon. If we also want to have the non-reduced variant now this makes this PR more complex than necessary.

Reply:
I can create a new pull request with the reduced variant. Before I do that I am going to try to figure out what broke when I merged develop with the re-meshing features into this branch. Ideally I would still keep the non reduced variant, since it has a lot nicer mathematical properties. 

For example if the coupled system is linear then the full QNWR will converge in Nd iteration, where d corresponds to the number of grid points in the interface and N to the number of time steps. If enough information from previous time steps is included this can be reduced to 2 iterations. Both of these statements do not hold for the reduced variant. I can also design linear systems for which the reduced variant converges very slowly. I have however not been able to find a linear system except for the non trivial ones where it does not converge to the correct solution.

Reply:
> For example if the coupled system is linear then the full QNWR will converge in Nd iteration, where d corresponds to the number of grid points in the interface and N to the number of time steps. If enough information from previous time steps is included this can be reduced to 2 iterations. Both of these statements do not hold for the reduced variant. I can also design linear systems for which the reduced variant converges very slowly. I have however not been able to find a linear system except for the non trivial ones where it does not converge to the correct solution.

Sounds to me like we should be able to show this for the heat equation? If this is possible, then I think looking at this question in another PR would be interesting.

Don't get me wrong: If the reduced variant has advantages over the non-reduced variant it we can definitely keep it. What you are mentioning sounds like we could use the heat equation as an end-to-end test for QNWI similar to how I used the heat equation with the manufactured solution to develop the time stepping. But I'm currently trying to close some boxes and not open new ones. That's the main reason I'm trying to get this PR merged rather sooner than later.

Reply:
IMO moving the non-reduced variant into a separate PR sounds unnecessarily complicated.
We have the implementation and we have integration tests here.

The only missing piece that we could do on top would be to repeat (some of) the tests of https://github.com/precice/precice/pull/2005#issuecomment-2425100108
@BenjaminRodenberg I assume you have this scripted. This should only be one config option to change and then start, no?

Reply:
> IMO moving the non-reduced variant into a separate PR sounds unnecessarily complicated.

The additional effort for factoring out the non-reduced QNWI should be negligible. My point here is that the non-reduced variant does not seem like a must-have to me and generally smaller PRs are easier to integrate, review and test than big ones. (This was also my motivation for opening https://github.com/precice/precice/pull/2107 and https://github.com/precice/precice/pull/2108)

At the moment there seems to be an integration problem with the current state of develop. Manual testing before this problem is fixed looks to me like a waste of time. We can discuss the right approach at the coding days but my opinion here is still that it makes more sense to reduce the feature set of this PR to get it merged soon. With all the things going on on develop at the moment regarding dynamic meshes we will get more work the longer we wait.

The scripts for running the tests are here: https://github.com/BenjaminRodenberg/tutorials/blob/dissertation-partitioned-heat-conduction-v3/partitioned-heat-conduction/doAccelerationStudy.py. Extending the scripts and running the tests is simple, but still requires one or two hours of work. I can also do this at the coding days. But as long as the PR is not final, I don't want to spend time here.